lib/spack/spack/test/cmd/install.py

```diff
@@ -376,3 +378,51 @@ def test_install_mix_cli_and_files(clispecs, filespecs, tmpdir):
 
     install(*args, fail_on_error=False)
     assert install.returncode == 0
+
+
+@pytest.mark.disable_clean_stage_check
+def test_cdash_report_concretization_error(tmpdir, mock_fetch, install_mockery,
+                                           capfd, conflict_spec):
+    # capfd interferes with Spack's capturing
+    with capfd.disabled():
+        with tmpdir.as_cwd():
+            try:
+                install(
+                    '--log-format=cdash',
+                    '--log-file=cdash_reports',
+                    conflict_spec)
+            except SpackError:
+                report_dir = tmpdir.join('cdash_reports')
+                assert report_dir in tmpdir.listdir()
+                report_file = report_dir.join('Update.xml')
+                assert report_file in report_dir.listdir()
+                content = report_file.open().read()
+                assert '<UpdateReturnStatus>Conflicts in concretized spec' \
+                    in content
+            else:
```

Comment:
I think you can avoid the complicated try/except/else logic if you use `with pytest.raises(SpackError): ...`.  It'll give a clearer error message when the exception doesn't occur, too.

Reply:
The test exits cleanly as soon as the exception is encountered when using `pytest.raises()`.  This prevents us from validating that the report was created successfully.  For example, the test still passes if I do something like this:

```python
with pytest.raises(SpackError):
    install(
        '--log-format=cdash',
        '--log-file=cdash_reports',
        conflict_spec)
    assert False
```

Reply:
@zackgalbreath Is there something preventing from doing:
```python
with tmpdir.as_cwd():
    with pytest.raises(SpackError):
        install(
            '--log-format=cdash',
            '--log-file=cdash_reports',
            conflict_spec)

    report_dir = tmpdir.join('cdash_reports')
    assert report_dir in tmpdir.listdir()
    ...
```
?

Reply:
Thanks for the nudge in the right direction.  This approach will work nicely.